project-base/assets/js/frontend/components/NewsletterSubscriptionForm.js

```diff
@@ -2,13 +2,19 @@ import Ajax from 'framework/common/utils/Ajax';
 import Register from 'framework/common/utils/Register';
 import Window from '../utils/Window';
 import Translator from 'bazinga-translator';
+import CustomizeBundle from 'framework/common/validation/customizeBundle';
 
 const subscriptionFormSelector = 'form[name="subscription_form"]';
 
 export default class NewsletterSubscriptionForm {
 
     ajaxSubmit (event, newsletterSubscriptionForm) {
         event.preventDefault();
+
+        if (CustomizeBundle.isFormValid($('form[name="subscription_form"]')) === false) {
```

Comment:
This is a very strange use case. We want to send form by ajax, so we add the submit event to the form (the second, because jsFormValidator adds the first event). So JsFormValidator validates the form and displays error window (event 1), followed by our event 2, which sends an ajax request (incorrectly because the form is not valid) and displays next error window. So I will submit ajax only after I check that the form is valid.

Reply:
I understand, thanks for the clarification.